app/service/agent_svc.py

```diff
@@ -137,17 +137,25 @@ def __init__(self, untrusted_timer):
         operation = (await data_svc.dao.get('core_operation', dict(id=op_id)))[0]
         while operation['state'] != operation_svc.op_states['RUNNING']:
             if operation['state'] == operation_svc.op_states['RUN_ONE_LINK']:
-                link_id = await data_svc.create('core_chain', link)
+                link_id = await self._create_link(link, data_svc)
                 await data_svc.dao.update('core_operation', 'id', op_id, dict(state=operation_svc.op_states['PAUSED']))
                 return link_id
             else:
                 await asyncio.sleep(30)
                 operation = (await data_svc.dao.get('core_operation', dict(id=op_id)))[0]
         link.pop('adversary_map_id')
-        return await data_svc.create('core_chain', link)
+        return await self._create_link(link, data_svc)
```

Comment:
good extraction of logic.
1 change- instead of passing data_svc as a parameter, can you make data_svc a module-level variable? so the init function would be:
```
self.data_svc = self.get_service('data_svc')
```
Then we can use it more naturally across the agent_svc class